lib/nosedrum/interactor.ex

```diff
@@ -0,0 +1,143 @@
+  alias Nostrum.Struct.Interaction
+
+  @callback_type_map %{
+    pong: 1,
+    channel_message_with_source: 4,
+    deferred_channel_message_with_source: 5,
+    deferred_update_message: 6,
+    update_message: 7
+  }
+
+  @flag_map %{
+    ephemeral?: 64
+  }
+
+  @type command_scope :: :global | Snowflake.t() | [Snowflake.t()]
+
+  @typedoc """
+  Defines a structure of commands, subcommands, subcommand groups, as
+  outlined in the [official documentation](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups)
+
+  **Note** that Discord only supports nesting 3 levels deep, like `command -> subcommand group -> subcommand`.
+
+  ## Example path:
+  ```elixir
+  %{
+    {"castle", MyApp.CastleCommand.description()} =>
+      %{
+        {"prepare", "Prepare the castle for an attack."} => [],
+        {"open", "Open up the castle for traders and visitors."} => [],
+        # ...
+      }
+  }
+  ```
+  """
+  @type application_command_path ::
+          %{
+            {group_name :: String.t(), group_desc :: String.t()} => [
+              application_command_path | [Nosedrum.ApplicationCommand.option()]
+            ]
+          }
+
+  @typedoc """
+  The name or pid of the Interactor process.
+  """
+  @type name_or_pid :: atom() | pid()
+
+  @doc """
+  Handle an Application Command invocation.
+
+  This callback should be invoked upon receiving an interaction via the `:INTERACTION_CREATE` event.
+
+  ## Example using `Nosedrum.Interactor.Dispatcher`:
+  ```elixir
+  # In your `Nostrum.Consumer` file:
+  def handle_event({:INTERACTION_CREATE, interaction, _ws_state}) do
+    IO.puts "Got interaction"
+    Nosedrum.Interactor.Dispatcher.handle_interaction(interaction)
+  end
+  ```
+  """
+  @callback handle_interaction(interaction :: Interaction.t(), name_or_pid) :: :ok
+
+  @doc """
+  Add a new command under the given name or application command path. Returns `:ok` if successful, and
+  `{:error, reason}` otherwise.
+
+  If the command already exists, it will be overwritten.
+  """
+  @callback add_command(
+              name_or_path :: String.t() | application_command_path,
+              command_module :: module,
+              scope :: command_scope,
+              name_or_pid
+            ) ::
+              :ok | {:error, reason :: String.t()}
+
+  @doc """
+  Remove the command under the given name or application command path. Returns `:ok` if successful, and
+  `{:error, reason}` otherwise.
+
+  If the command does not exist, no error should be returned.
+  """
+  @callback remove_command(
+              name_or_path :: String.t() | application_command_path,
+              command_id :: Nostrum.Snowflake.t(),
+              scope :: command_scope,
+              name_or_pid
+            ) ::
+              :ok | {:error, reason :: String.t()}
+
+  @doc """
+  Responds to an Interaction with the values in the given `t:Nosedrum.ApplicationCommand.response()`. Returns `{:ok}` if
```

Comment:
To link properly here, I think we need to add `/0` instead of `()`:

```suggestion
  Responds to an Interaction with the values in the given `t:Nosedrum.ApplicationCommand.response/0`. Returns `{:ok}` if
```